Kandev backend not being built correctly - "go-sqlite3 requires cgo to work"

### Description

On a MacBook Pro M4 Pro, in a Docker/devcontainer environment, if I try to run `make dev`, the following error appears:

```
Running kandev in development mode...
2026-04-14T09:39:03.357Z        INFO    logger/logger.go:188    Starting Kandev (unified mode)...       {"db_path": "/workspaces/kandev/apps/backend/kandev.db"}
2026-04-14T09:39:03.357Z        INFO    logger/logger.go:188    Initializing Task Service...
2026-04-14T09:39:03.357Z        INFO    logger/logger.go:188    Database initialized (single-writer pool)       {"db_path": "/workspaces/kandev/apps/backend/kandev.db", "db_driver": "sqlite"}
2026-04-14T09:39:03.357Z        ERROR   logger/logger.go:198    Failed to initialize repositories       {"error": "failed to initialize schema: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub"}
```

I do have `gcc` on my environment. The build just doesn't use it 😅 

### Fix Suggestion

Since the backend always requires `cgo` in order to build properly, consider setting `CGO_ENABLED=1` for the backend build. 
It's preferable for the build to crash over missing `gcc`, rather than building successfully, but stubbing `go-sqlite3` 😅

On a side note, if I set `go env CGO_ENABLED=1` and then `make dev`, I get the following error:

```
$ make dev
Building Linux agentctl for remote executors...
make[1]: Entering directory '/workspaces/kandev/apps/backend'
Building agentctl for linux/amd64...
GOOS=linux GOARCH=amd64 go build -v -ldflags "-s -w -X main.Version=v0.32-2-g3b630132 -X main.Commit=3b630132 -X main.BuildTime=2026-04-14T10:36:05Z" -o bin/agentctl-linux-amd64 ./cmd/agentctl
# runtime/cgo
cc: error: unrecognized command-line option '-m64'
make[1]: *** [Makefile:60: build-agentctl-linux] Error 1
make[1]: Leaving directory '/workspaces/kandev/apps/backend'
make: *** [Makefile:110: dev] Error 2
```

On the project root's `Makefile:107-111`, the following code decides if it should run the `build-agentctl-linux`, which builds for the `amd64` architecture:

```
dev:
ifneq ($(shell uname -s)/$(shell uname -m),Linux/x86_64)
	@echo "Building Linux agentctl for remote executors..."
	@$(MAKE) -C $(BACKEND_DIR) build-agentctl-linux
endif
	@echo "Launching via CLI (auto ports)..."
	@cd $(APPS_DIR) && $(PNPM) -C cli dev -- dev
```

However, because the devcontainer is running on an M4 Pro, the architecture is `arm64`:

```
$ echo $(uname -s)/$(uname -m)
Linux/aarch64
```

But because `CGO_ENABLED=1`, the build then errors out, because it attempts to build with gcc, and `-m64` is not a valid flag for the ARM gcc. Thus, I can't set the `CGO_ENABLED` variable globally.

So this also points out a separate bug in the root `Makefile`, but I think that the most important issue here is `go-sqlite3` possibly being stubbed on build silently 😅 At least if the flag is only enabled for the backend build, then running `make build` or `make dev` on root succeeds anyway, since the C compiler isn't used for the Linux agentctl build.

--- a/apps/backend/Makefile
+++ b/apps/backend/Makefile
@@ -40,7 +40,7 @@ build-mock-agent:
 build: build-agentctl build-mock-agent
 	@echo "Building $(BINARY_NAME)..."
 	@$(MKDIR) $(BUILD_DIR)
-	$(GO) build $(GOFLAGS) -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME) ./cmd/kandev
+	CGO_ENABLED=1 $(GO) build $(GOFLAGS) -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME) ./cmd/kandev
 
 ## Build all binaries (unified + individual services + agentctl)
 build-all: build build-agentctl
@@ -84,7 +84,7 @@ start-debug: build
 ## Run tests
 test:
 	@echo "Running tests..."
-	$(GO) test ./...
+	CGO_ENABLED=1 $(GO) test ./...
 
 ## Run E2E tests (real agents, costs money — manual only)
 ## Set agent commands via env vars: E2E_CLAUDE_CODE_COMMAND, E2E_AMP_COMMAND

--- a/apps/backend/Makefile
+++ b/apps/backend/Makefile
@@ -103,7 +103,7 @@ test-sprites-e2e: build-agentctl-linux
 ## Run tests with coverage
 test-coverage:
 	@echo "Running tests with coverage..."
-	$(GO) test -v -coverprofile=coverage.out ./...
+	CGO_ENABLED=1 $(GO) test -v -coverprofile=coverage.out ./...
 	$(GO) tool cover -html=coverage.out -o coverage.html
 
 ## Run linter (requires golangci-lint)
@@ -135,10 +135,10 @@ deps:
 build-cross:
 	@echo "Building for multiple platforms..."
 	@$(MKDIR) $(BUILD_DIR)
-	GOOS=linux GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-linux-amd64 ./cmd/kandev
-	GOOS=darwin GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-darwin-amd64 ./cmd/kandev
-	GOOS=darwin GOARCH=arm64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-darwin-arm64 ./cmd/kandev
-	GOOS=windows GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-windows-amd64.exe ./cmd/kandev
+	CGO_ENABLED=1 GOOS=linux GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-linux-amd64 ./cmd/kandev
+	CGO_ENABLED=1 GOOS=darwin GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-darwin-amd64 ./cmd/kandev
+	CGO_ENABLED=1 GOOS=darwin GOARCH=arm64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-darwin-arm64 ./cmd/kandev
+	CGO_ENABLED=1 GOOS=windows GOARCH=amd64 $(GO) build -ldflags "$(LDFLAGS)" -o $(BUILD_DIR)/$(BINARY_NAME)-windows-amd64.exe ./cmd/kandev
 
 ## Show help
 help:

